shared/src/utils/propositionalLogicMinimize.ts

```diff
@@ -0,0 +1,324 @@
+import {
+  BinaryOperatorType,
+  Literal,
+  NormalForm,
+  Operator,
+  SyntaxTreeNodeType,
+  TruthTable,
+} from "@shared/utils/propositionalLogic.ts"
+
+/**
+ * During the Quine–McCluskey algorithm, the datatype for a truthTable can be a BinaryMix
+ * 0 --> false; 1 --> true; x --> true/false
+ */
+type BinaryMix = 0 | 1 | "x"
+
+/**
+ * This class gets the minimal normal form of a given expression.
+ * - DNF (Disjunctive Normal Form)
+ * - CNF (Conjunctive Normal
+ * - The minimal normal form is the minimal number of literals needed
+ */
+export class GetMinimalNormalForm {
+  private readonly expr: SyntaxTreeNodeType
+  private readonly cnfExpr: SyntaxTreeNodeType
+  private readonly dnfExpr: SyntaxTreeNodeType
+
+  constructor(expr: SyntaxTreeNodeType) {
+    this.expr = expr
+    this.dnfExpr = this.getMinimalForm("DNF")
+    this.cnfExpr = this.getMinimalForm("CNF")
+  }
+
+  /**
+   * Returns the expression
+   * @param form - which form of expression
+   */
+  get(form: NormalForm | "base"): SyntaxTreeNodeType {
+    if (form === "DNF") return this.dnfExpr
+    if (form === "CNF") return this.cnfExpr
+    return this.expr
+  }
+
+  /**
+   * Returns the string representation of the SyntaxTreeNode
+   * Shortcut for x.get().toString()
+   * @param latex - whether to return the latex representation
+   * @param form - which form returned
+   */
+  toString(latex: boolean = false, form: NormalForm | "base"): string {
```

Comment:
This function is currently not used. Instead `x.get(form).toString()` is used, which is totally fine. I think it is not really an overhead das requires a shortcut. If the class is changed in a way, that the form is supplied to the constructor, `get()` does not need the parameter anymore, so the need for a shortcut gets even smaller. I would just remove the function.